distiller/audit_search/forms.py

```diff
@@ -1,110 +1,119 @@
+    ('30', 'Equal Employment Opportunity Commission'),
+    ('95', 'Executive Office of the President'),
+    ('32', 'Federal Communications Commission'),
+    ('83', 'Federal Emergency Management Agency'),
+    ('33', 'Federal Maritime Commission'),
+    ('34', 'Federal Mediation and Conciliation Service'),
+    ('18', 'Federal Reserve System'),
+    ('36', 'Federal Trade Commission'),
+    ('39', 'General Services Administration'),
+    ('40', 'Government Printing Office'),
+    ('03', 'Institute of Museum and Library Services'),
+    ('04', 'Inter-American Foundation'),
+    ('61', 'International Trade Commission'),
+    ('90', 'Japan U.S. Friendship Commission'),
+    ('09', 'Legal Services Corporation'),
+    ('42', 'Library of Congress'),
+    ('43', 'National Aeronautics & Space Administration'),
+    ('89', 'National Archives & Records Administration'),
+    ('92', 'National Council on Disability'),
+    ('44', 'National Credit Union Administration'),
+    ('05', 'National Endowment for the Arts'),
+    ('06', 'National Endowment for the Humanities'),
+    ('45', 'National Foundation on the Arts and the Humanities'),
+    ('68', 'National Gallery of Art'),
+    ('46', 'National Labor Relations Board'),
+    ('47', 'National Science Foundation'),
+    ('77', 'Nuclear Regulatory Commission'),
+    ('07', 'Office of National Drug Control Policy'),
+    ('27', 'Office of Personnel Management'),
+    ('70', 'Overseas Private Investment Corporation'),
+    ('08', 'Peace Corps'),
+    ('86', 'Pension Benefit Guaranty Corporation'),
+    ('22', 'Postal Service'),
+    ('53', "President's Committee on Employment of People with Disabilities"),
+    ('57', 'Railroad Retirement Board'),
+    ('85', 'Scholarship Foundations'),
+    ('58', 'Securities and Exchange Commission'),
+    ('59', 'Small Business Administration'),
+    ('60', 'Smithsonian Institution'),
+    ('96', 'Social Security Administration'),
+    ('62', 'Tennessee Valley Authority'),
+    ('98', 'U.S. Agency for International Development'),
+    ('82', 'United States Information Agency'),
+    ('91', 'United States Institute of Peace'),
+    ('99', 'Miscellaneous'),
+)
 
+AGENCIES_BY_PREFIX = defaultdict(list)
+for choice in AGENCY_CHOICES:
+    AGENCIES_BY_PREFIX[choice[0]].append(choice[1])
 
-def _is_valid_agency_prefix(agency_prefix):
-    ''' Given a two-digit prefix a la the Federal Audit Clearinghouse, return
-    True if it's one of the prefixes that the FAC lists as a "federal agency
-    prefix."
 
-    Implementing this as a dict lookup instead of a range check because not
-    every two-digit combination between 00 and 99 is actually valid (i.e., listed
-    in FAC).
-    '''
-    return agency_prefix in AGENCIES_BY_PREFIX
+class AgencySelectionForm(forms.Form):
+    agency = forms.ChoiceField(choices=AGENCY_CHOICES)
+    sub_agency = forms.ChoiceField(required=False)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
Do we need this?

Reply:
Yes - we pass data into the constructor that needs to be registered with the parent class.